Simple workaround to prevent crash of host ldd when it try to find/check
shared libraries from different architecture.

diff -uBp uClibc-0.9.30.1/utils/ldd.c uClibc/utils/ldd.c
--- uClibc-0.9.30.1/utils/ldd.c	2008-10-29 22:40:21.000000000 +0300
+++ uClibc/utils/ldd.c	2009-09-06 12:43:18.000000000 +0400
@@ -519,6 +519,8 @@ static int add_library(ElfW(Ehdr) *ehdr,
 	for (cur = lib_list; cur; cur = cur->next) {
 		/* Check if this library is already in the list */
 		tmp1 = tmp2 = cur->name;
+		if (!cur->name)
+			continue;
 		while (*tmp1) {
 			if (*tmp1 == '/')
 				tmp2 = tmp1 + 1;
@@ -595,6 +597,8 @@ static struct library *find_elf_interpre
 		}
 		for (cur = lib_list; cur; cur = cur->next) {
 			/* Check if this library is already in the list */
+			if (!tmp1 || !cur->name)
+				return NULL;
 			if (strcmp(cur->name, tmp1) == 0) {
 				/*printf("find_elf_interpreter is replacing '%s' (already in list)\n", cur->name); */
 				newlib = cur;
